Core/src/org/sleuthkit/autopsy/centralrepository/eventlisteners/CaseEventListener.java

```diff
@@ -249,91 +309,87 @@ private BlackboardTagTask(CentralRepository db, PropertyChangeEvent evt) {
 
         @Override
         public void run() {
-            if (!CentralRepository.isEnabled()) {
+            if (!CentralRepository.isEnabled())
                 return;
-            }
 
-            Content content;
-            BlackboardArtifact bbArtifact;
-            TskData.FileKnown knownStatus;
-            String comment;
             if (Case.Events.valueOf(event.getPropertyName()) == Case.Events.BLACKBOARD_ARTIFACT_TAG_ADDED) {
-                // For added tags, we want to change the known status to BAD if the 
-                // tag that was just added is in the list of central repo tags.
-                final BlackBoardArtifactTagAddedEvent tagAddedEvent = (BlackBoardArtifactTagAddedEvent) event;
-                final BlackboardArtifactTag tagAdded = tagAddedEvent.getAddedTag();
-
-                if (TagsManager.getNotableTagDisplayNames().contains(tagAdded.getName().getDisplayName())) {
-                    content = tagAdded.getContent();
-                    bbArtifact = tagAdded.getArtifact();
-                    knownStatus = TskData.FileKnown.BAD;
-                    comment = tagAdded.getComment();
-                } else {
-                    // The added tag isn't flagged as bad in central repo, so do nothing
-                    return;
-                }
-            } else { //BLACKBOARD_ARTIFACT_TAG_DELETED
-                Case openCase;
-                try {
-                    openCase = Case.getCurrentCaseThrows();
-                } catch (NoCurrentCaseException ex) {
-                    LOGGER.log(Level.SEVERE, "Exception while getting open case.", ex);
-                    return;
-                }
-                // For deleted tags, we want to set the file status to UNKNOWN if:
-                //   - The tag that was just removed is notable in central repo
-                //   - There are no remaining tags that are notable 
-                final BlackBoardArtifactTagDeletedEvent tagDeletedEvent = (BlackBoardArtifactTagDeletedEvent) event;
-                long contentID = tagDeletedEvent.getDeletedTagInfo().getContentID();
-                long artifactID = tagDeletedEvent.getDeletedTagInfo().getArtifactID();
+                handleTagAdded((BlackBoardArtifactTagAddedEvent) event);
+            } 
+            else { 
+                handleTagDeleted((BlackBoardArtifactTagDeletedEvent) event);
+            }
+        } // BLACKBOARD_ARTIFACT_TAG_ADDED, BLACKBOARD_ARTIFACT_TAG_DELETED
+        
+        private void handleTagDeleted(BlackBoardArtifactTagDeletedEvent tagDeletedEvent) {
```

Comment:
See remarks on the file tag added/deleted event handling; artifact tag added/deleted event handling is similar.